File: mockmate-backend/src/main/resources/runners/java/runner_int_array+int_array+int.java
import java.util.*;
import java.util.regex.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        StringBuilder sb = new StringBuilder();
        while (sc.hasNextLine()) sb.append(sc.nextLine()).append(" ");
        String fullInput = sb.toString();

        try {
            List<int[]> arrays = new ArrayList<>();
            Matcher mBracket = Pattern.compile("\\[(.*?)\\]").matcher(fullInput);
            int lastEnd = 0;
            while(mBracket.find()) {
                String arrayPart = mBracket.group(1);
                arrays.add(parseArrayString(arrayPart));
                lastEnd = mBracket.end();
            }

            String remaining = fullInput.substring(lastEnd);
            Matcher mInt = Pattern.compile("-?\\d+").matcher(remaining);
            int k = 0;
            if(mInt.find()) k = Integer.parseInt(mInt.group());

            int[] arr1 = arrays.size() > 0 ? arrays.get(0) : new int[0];
            int[] arr2 = arrays.size() > 1 ? arrays.get(1) : new int[0];

            Solution sol = new Solution();
            Object result = sol.{{methodSignature}}(arr1, arr2, k);
            if(result instanceof int[]) System.out.println(Arrays.toString((int[])result));
            else if(result instanceof Boolean) System.out.println(result.toString().toLowerCase());
            else System.out.println(result);
        } catch (Exception e) {
            e.printStackTrace();
            System.exit(1);
        } finally {
            sc.close();
        }
    }

    static int[] parseArrayString(String s) {
        s = s.replace(",", " ").trim();
        if (s.isEmpty()) return new int[0];
        String[] parts = s.split("\\s+");
        List<Integer> list = new ArrayList<>();
        for (String p : parts) {
            try { list.add(Integer.parseInt(p)); } catch(Exception ignored){}
        }
        int[] res = new int[list.size()];
        for (int i = 0; i < list.size(); i++) res[i] = list.get(i);
        return res;
    }
}

{{USER_CODE}}

File: mockmate-backend/src/main/resources/runners/python/runner_int_array+int_array+int.py
import sys
import json
import re

def main():
    input_data = sys.stdin.read().strip()
    if not input_data:
        return

    arrays = []
    depth = 0
    start = -1
    last_end = 0
    for i, c in enumerate(input_data):
        if c == '[':
            if depth == 0: start = i
            depth += 1
        elif c == ']':
            depth -= 1
            if depth == 0 and start != -1:
                try:
                    arrays.append(json.loads(input_data[start:i+1]))
                except:
                    pass
                last_end = i + 1
                start = -1

    remaining = input_data[last_end:]
    nums = re.findall(r'-?\d+', remaining)
    k = int(nums[0]) if nums else 0

    arr1 = arrays[0] if len(arrays) > 0 else []
    arr2 = arrays[1] if len(arrays) > 1 else []

    sol = Solution()
    result = sol.{{methodSignature}}(arr1, arr2, k)

    if isinstance(result, bool):
        print(str(result).lower())
    else:
        print(json.dumps(result).replace(' ', ''))

{{USER_CODE}}

if __name__ == '__main__':
    main()
